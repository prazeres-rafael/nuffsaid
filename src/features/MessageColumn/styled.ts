import styled from "styled-components";

export const ButtonClear = styled.a({
  cursor: "pointer",
});

export const Subtitle = styled.p({
  fontSize: 20,
  fontWeight: 800,
});
