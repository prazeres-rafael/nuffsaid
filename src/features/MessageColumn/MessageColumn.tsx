import React from "react";
import { Message } from "../../Api";
import BoxPriority from "../../components/BoxPriority";
import { Subtitle } from "./styled";
import { Grid } from "@mui/material";

type Props = {
  messages: Message[];
};

export default function MessageColumn({ messages }: Props): React.ReactElement {
  return (
    <>
      <Grid item xs={4}>
        <Subtitle>Error Type 1</Subtitle>
        <BoxPriority priority={0} messages={messages} />
      </Grid>
      <Grid item xs={4}>
        <Subtitle>Warning Type 2</Subtitle>
        <BoxPriority priority={1} messages={messages} />
      </Grid>
      <Grid item xs={4}>
        <Subtitle>Info Type 3</Subtitle>
        <BoxPriority priority={2} messages={messages} />
      </Grid>
    </>
  );
}
