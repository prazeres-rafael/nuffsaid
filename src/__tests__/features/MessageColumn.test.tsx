import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import MessageColumn from "../../features/MessageColumn/MessageColumn";

const messages = [
  { message: "Et aut quo laboriosam at ut.", priority: 0 },
  { message: "Dolor quo est expedita nostrum corrupti.", priority: 1 },
  { message: "Recusandae iure et.", priority: 2 },
];

describe("MessageColumn", () => {
  it("should match the snapshot", () => {
    const { container } = render(<MessageColumn messages={messages} />);
    expect(container).toMatchSnapshot();
  });
});
