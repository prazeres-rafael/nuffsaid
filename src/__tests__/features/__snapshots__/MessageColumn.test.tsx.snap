// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageColumn should match the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
  >
    <p
      class="sc-dkPtRN jLMnWC"
    >
      Error Type 1
    </p>
    <p>
      Count: 
      1
    </p>
    <div
      class="sc-bdvvtL cVJmSZ"
    >
      Et aut quo laboriosam at ut.
      <a
        class="sc-gsDKAQ ertJLs"
        data-testid="clear-button"
      >
        Clear
      </a>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
  >
    <p
      class="sc-dkPtRN jLMnWC"
    >
      Error Type 2
    </p>
    <p>
      Count: 
      1
    </p>
    <div
      class="sc-bdvvtL dRCsmw"
    >
      Dolor quo est expedita nostrum corrupti.
      <a
        class="sc-gsDKAQ ertJLs"
      >
        Clear
      </a>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
  >
    <p
      class="sc-dkPtRN jLMnWC"
    >
      Error Type 3
    </p>
    <p>
      Count: 
      1
    </p>
    <div
      class="sc-bdvvtL VebcU"
    >
      Recusandae iure et.
      <a
        class="sc-gsDKAQ ertJLs"
      >
        Clear
      </a>
    </div>
  </div>
</div>
`;
